app/controlplane/internal/service/service.go

```diff
@@ -181,19 +182,29 @@ func (s *service) authorizeResource(ctx context.Context, op *authz.Policy, resou
 }
 
 // userHasPermissionOnProject is a helper method that checks if a policy can be applied to a project. It looks for a project
-// by name and ensures that the user has a role that allows that specific operation in the project.
+// by name in the given organization and ensures that the user has a role that allows that specific operation in the project.
 // check authorizeResource method
-func (s *service) userHasPermissionOnProject(ctx context.Context, orgID string, pName string, policy *authz.Policy) error {
+// if it doesn't return an error, it means that the user has the permission and the project is returned
+func (s *service) userHasPermissionOnProject(ctx context.Context, orgID string, pName string, policy *authz.Policy) (*biz.Project, error) {
+	p, err := s.projectUseCase.FindProjectByReference(ctx, orgID, &biz.EntityRef{Name: pName})
+	if err != nil {
+		return nil, handleUseCaseErr(err, s.log)
+	}
+
+	if p == nil {
```

Comment:
I think this case is correctly handled in the data layer already. There is no case where `p==nil` and `err==nil`

Reply:
You are right, the issue was that I was not sending back the project, Good catch